docs/dev/design/new-search-api.rst

```diff
@@ -200,43 +202,96 @@ Examples
+so a user can search all their projects easily.
+We could allow just ``@me`` to start.
+
+Facets
+~~~~~~
+
+We can keep the facets, but they would be a little different,
+since with the new syntax we need to specify a project in order to search for
+a version, i.e, we can't search all ``latest`` versions of all projects.
+
+By default we will use/show the ``project`` facet,
+and after the user has filtered by a project,
+we will use/show the ``version`` facet.
+
+If the user searches more than one project,
+things get complicated, should we keep showing the ``version`` facet?
+If clicked, should we change the version on all the projects?
+
+If that is too complicated to explain/implement,
+we should be fine by just supporting the ``project``
+facet for now.
+
+Backwards compatibility
+~~~~~~~~~~~~~~~~~~~~~~~
+
+We should be able to keep the old URLs working in the global search,
+but we could also just ignore the old syntax, or transform
+the old syntax to the new one and redirect the user to it,
+for example ``?q=test&project=docs&version=latest``
+would be transformed to ``?q=test project:docs/latest``.
 
 Future features
 ---------------
 
 - Allow searching on several versions of the same project
   (the API response is prepared to support this).
+- Allow searching on all versions of a project easily,
+  with a syntax like ``project:docs/*`` or ``project:docs/@all``.
 - Allow specify the type of search:
 
   - Multi match (query as is)
   - Simple query string (allows using the ES query syntax)
   - Fuzzy search (same as multi match, but with with fuzziness)
 
-Questions / pending decisions
------------------------------
-
-Integration with the dashboard search.
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The search API and the search from the dashboard
-use the same backend, but they are used in a different way.
-
-The dashboard search by default searches on all projects (.org),
-or all project the current user has access to (.com).
-And the API search searches on explicitly given projects.
-
-The dashboard search allows filtering by version and role,
-the API search allows filtering only by version (and it's required).
-
-The dashboard search makes use of filters in order to return
-the number of results from other versions/roles.
-Is this feature useful? It could slow down the response.
-Searching several versions at the same time could be a better replace?
-
-The dashboard search can be used to search for projects
-by their name and description.
-The API search doesn't support this.
-Is this feature useful? Should we implement
-this as a way to filter projects from https://readthedocs.org/dashboard/ instead?
-This will be using just https://docs.djangoproject.com/en/4.0/ref/contrib/postgres/search/
-or ``contains``.
+- Add the ``organization`` filter,
+  so users can search by all projects that belong
+  to an organization.
+  Would we show results of all versions
+  or just the default version?
```

Comment:
If we can default to the default version, but have that be different for each project, that would probably be the most obvious option. It's really hard UX returning search results from multiple versions of the same page.